### PR TITLE
Support reading `bootstrap_nodes.yaml`

### PR DESCRIPTION
**Pull request by etan-status**
`bootstrap_nodes.txt` is retired in lieu of `bootstrap_nodes.yaml`, start reading `.yaml` format (similar to `.enr`).

### CODE REVIEW

File: beacon_chain/networking/network_metadata.nim [hunk 1]
```diff
@@ -126,7 +144,8 @@ proc loadEth2NetworkMetadata*(
       deployBlockPath = path & "/deploy_block.txt"
       depositContractBlockPath = path & "/deposit_contract_block.txt"
       depositContractBlockHashPath = path & "/deposit_contract_block_hash.txt"
-      bootstrapNodesPath = path & "/bootstrap_nodes.txt"
+      bootstrapNodesLegacyPath = path & "/bootstrap_nodes.txt"  # <= Dec 2024
```
Comment (tersec) [inline beacon_chain/networking/network_metadata.nim:147]: This is a good way to more safely transition especially given (a) a potentially early December release and (b) possible legacy-format-metadata custom networks people might have. Worth considering at some point removing the older versions. But or this PR, sure.

File: beacon_chain/networking/network_metadata.nim [hunk 2]
```diff
@@ -91,23 +91,41 @@ type
...
+proc readBootstrapNodes(path: string): seq[string] {.raises: [IOError].} =
   # Read a list of ENR values from a YAML file containing a flat list of entries
+  var res: seq[string]
   if fileExists(path):
-    splitLines(readFile(path)).
-      filterIt(it.startsWith("enr:")).
-      mapIt(it.strip())
-  else:
-    @[]
+    for line in splitLines(readFile(path)):
```
Comment (tersec) [inline beacon_chain/networking/network_metadata.nim:98]: This isn't really a YAML parser, this is a parser for that specific still-basically-line-oriented format. Is that enshrined as what's necessary, or is it legal now to have true general-purpose YAML files?

We do have `NimYAML` if necessary.

File: beacon_chain/networking/network_metadata.nim [hunk 3]
```diff
@@ -91,23 +91,41 @@ type
...
+proc readBootEnr(path: string): seq[string] {.raises: [IOError].} =
   # Read a list of ENR values from a YAML file containing a flat list of entries
+  var res: seq[string]
   if fileExists(path):
-    splitLines(readFile(path)).
-      filterIt(it.startsWith("- enr:")).
-      mapIt(it[2..^1].strip())
-  else:
-    @[]
+    for line in splitLines(readFile(path)):
```
Comment (tersec) [inline beacon_chain/networking/network_metadata.nim:115]: This isn't really a YAML parser, this is a parser for that specific still-basically-line-oriented format. Is that enshrined as what's necessary, or is it legal now to have true general-purpose YAML files?

We do have `NimYAML` if necessary.

Comment (etan-status) [reply beacon_chain/networking/network_metadata.nim:115]: This is in line with how we parse `config.yaml` for network configurations. We kinda parse what exists in the wild; the `.yaml` extension for both the `boot_enr.yaml` (previously existed, and did not work on Chiado file) and the new `bootstrap_nodes.yaml` (same format seems to work) suggests that this is general purpose YAML in theory.

`NimYAML` so far is solely used in testing, and if we add it to production it should be audited accordingly. For now, I think going with the existing parser (plus the fix for Chiado which quotes strings, which is valid in YAML) is probably alright.

**Pull request merged**